feat: add `fft/base/fftpack/ndarray/drffti` (#13919)
stdlib-js/stdlib: lib/node_modules/@stdlib/fft/base/fftpack/ndarray/float64/rffti/README.md

```

#### rffti( arrays )

Initializes a workspace array for performing a real-valued Fourier transform on a one-dimensional double-precision floating-point ndarray.

```javascript
var Float64Vector = require( '@stdlib/ndarray/vector/float64' );
var scalar2ndarray = require( '@stdlib/ndarray/from-scalar' );
var Slice = require( '@stdlib/slice/ctor' );
var slice = require( '@stdlib/ndarray/slice' );

var N = 8;
var len = scalar2ndarray( N, {
    'dtype': 'int32'
});

var w = new Float64Vector( ( 2*N ) + 34 );

var out = rffti( [ w, len ] );
// returns <ndarray>

var bool = ( out === w );
// returns true

var twiddleFactors = slice( w, new Slice( N, 2*N ) );
// returns <ndarray>[ ~0.707, ~0.707, 0, 0, 0, 0, 0, 0 ]

var factors = slice( w, new Slice( 2*N, ( 2*N ) + 4 ) );
// returns <ndarray>[ 8, 2, 2, 4 ]
```

The function has the following parameters:

-   **arrays**: array-like object containing the following ndarrays:

    -   a one-dimensional input ndarray.
    -   a zero-dimensional ndarray containing the length of the sequence to transform.

</section>

<!-- /.usage -->

<section class="notes">

## Notes

> -   The workspace array is divided into three sections:

Reviewer: ```suggestion
-   Let `N` equal the number of elements in the input ndarray. The input ndarray is divided into three sections:
```

    ```text
              size = N                 N            2+ceil(log2(N)/2)
                   ↓                   ↓                  ↓
        | scratch / workspace | twiddle factors | radix factor table |
                   ↑                   ↑                  ↑
    i = 0         ...         N       ...       2N       ...
    ```

    -   **scratch/workspace**: used as a scratch space when performing transforms. This section is not updated during initialization.
    -   **twiddle factors**: a table of reusable complex-exponential constants stored as cosine/sine pairs.
    -   **radix factor table**: a table containing the sequence length `N`, the number of factors into which `N` was decomposed, and the individual integer radix factors.

> -   In general, a workspace array should have `2N + 34` indexed elements (as `log2(N)/2 ≤ 32` for all `2^64`). During initialization, only the sections for storing twiddle factors and the factorization of `N` are updated.

Reviewer: ```suggestion
-   In general, an input ndarray should have `2N + 34` indexed elements (as `log2(N)/2 ≤ 32` for all `2^64`). During initialization, only the sections for storing twiddle factors and the factorization of `N` are updated.
```

-   The radix factor table is comprised as follows:

    ```text
    | sequence_length | number_of_factors | integer_factors |
    ```

> -   If `N` equals `1`, the function returns early without modifying the workspace, as a single data point is its own Fourier transform.

Reviewer: ```suggestion
-   If `N` equals `1`, the function returns early without modifying the input ndarray, as a single data point is its own Fourier transform.
```
